rfc-117-containerised-runtime.md

```diff
@@ -0,0 +1,68 @@
+# RFC 000: Use a containerised runtime
+
+## Summary
+
+Use a containerised runtime for hosting GOV.UK applications. GOV.UK applications currently run on virtual machines running Ubuntu Trusty Tahr. There is currently an extended support contract for Ubuntu Trusty which will end in April 2022. Rather than upgrading Ubuntu on the virtual machines GOV.UK should change the application runtime to a containerised platform.
+
+## Problem
+
+GOV.UK currently hosts the majority of its applications and services using virtual machines running Ubuntu 14.04 LTS (Trusty Tahr).
+
+Trusty is out of date, it was released in April 2014 and standard support ended in April 2019. At this time GOV.UK signed up for [Extended Security Maintenance](https://ubuntu.com/esm) (ESM) which provides continued security fixes for high and critical common vulnerabilities and exposures (CVEs) for the [most commonly used packages](https://wiki.ubuntu.com/SecurityTeam/ESM/14.04#A14.04_Infrastructure_ESM_Packages). ESM will end in April 2022.
+
+Running an old operating system is preventing GOV.UK from upgrading key components of our technology stack (MongoDB, Puppet, Python, ...).
+
+Upgrading Ubuntu in the existing infrastructure is difficult for a number of reasons, including but not limited to:
```

Comment:
If I remember correctly one of the blockers (if not the main one) of upgrading to Xenial was that Icinga wouldn't run on Xenial. Not sure if that is still an issue.

Reply:
It looks like Icinga has [Xenial binaries](https://packages.icinga.com/ubuntu/dists/icinga-xenial/) now, but I don't know if there was another reason why it wouldn't run - perhaps we'd need to do a big Icinga upgrade to use those binaries?

I don't think we need to call out every blocker in the RFC though - it will end up being huge!